src/lib/common/payload_uart.cpp

```diff
@@ -188,6 +287,26 @@ SerialHandle_t uart_handle = {
     .postTxCb = NULL,
 };
 
+// This task exectutes the _postTxFunction.
+//  If transactions are being used, it can be triggered by
+//  endTransaction or UART Tx complete TC ISR -> pluartPostTxCallback, whichever occurs first.
+static void postTxCBTask(void *pvParameters) {
+  (void)pvParameters; // Mark 'pvParameters' as intentionally unused
+
+  for (;;) {
+    if (xSemaphoreTake(_postTxSemaphore, portMAX_DELAY) == pdTRUE) {
+      taskENTER_CRITICAL();
+      if (_postTxFunctionPending) {
+        _postTxFunctionPending = false;
+        taskEXIT_CRITICAL();
+        _postTxFunction(); // Safely call the function
+      } else {
+        taskEXIT_CRITICAL();
+      }
+    }
+  }
+}
+
```

Comment:
I need to put some more thought into this, but I think there is a path where we don't have the `postTxCbTask` and instead make more use of the `preTxCb` and `postTxCb` that are called in serial.c. 

Perhaps we can use a counting semaphore, such that each preTxCb will take(increment) the semaphore and then each postTxCb will give(decrement) the semaphore. This can only work so long as the serial tx task is higher priority than the task that calls serialTx, thus the Tx event in the serial tx queue should be processed (and increment the semaphore)  before we reach the endTransaction. 

Another possibility is that the pluart::write function increments the semaphore, therefore the user task (or other task) will increment the semaphore, and then the postTxCb will give it back.

Then we can wait in `endTransaction` until the counting semaphore is 0 and then do the post tx function stuff.

Reply:
I'm mainly trying to think of ways to avoid the need of another task and keep the task interactions isolated to between the user_task (or whoever is calling the pluart functions) and the serial task.

Reply:
@victorsowa12 The downside would be some blocking delay in the user task until all Tx are all complete, right? That would probably be acceptable.

Reply:
Yes correct, you would need some sort of polling loop in `endTransaction` like:
```
end Transaction() {
  while (semaphoreGetCount != 0) {
    delay(1);
  }
  // do end of tx stuff
}
```

Reply:
It would prevent someone from doing this:
```
start();
write();
end();  ---> if this is handled in another task then the next start may overlap during the delay this end may have
start();
write();
end();
```

Reply:
And then yeah any _postTxFunction delays would have to occur in the user context as well.